src/ucs/memory/memtype_cache.c

```diff
@@ -69,16 +69,16 @@ static void ucs_memtype_cache_pgt_dir_release(const ucs_pgtable_t *pgtable,
  * - Lock must be held in write mode
  * - start, end must be aligned to page size
  */
-static void ucs_memtype_cache_insert(ucs_pgt_addr_t start, ucs_pgt_addr_t end,
+static void ucs_memtype_cache_insert(ucs_memtype_cache_t *memtype_cache,
+                                     ucs_pgt_addr_t start, ucs_pgt_addr_t end,
                                      const ucs_memory_info_t *mem_info)
 {
-    ucs_memtype_cache_t *memtype_cache = ucs_memtype_cache_get_global();
     ucs_memtype_cache_region_t *region;
     ucs_status_t status;
     int ret;
 
     if (memtype_cache == NULL) {
-        ucs_error("no memtype_cache instance found");
+        ucs_trace("no memtype_cache instance found");
```

Comment:
now this check does not make sense because memtype_cache is passed as a parameter to the function, better remove